src/gen10_hevc_enc_common.c

```diff
@@ -0,0 +1,1398 @@
+        }
+    }
+}
+
+VAStatus
+gen10_hevc_enc_ensure_surface(VADriverContextP ctx,
+                              struct object_surface *obj_surface,
+                              int bit_depth_minus8,
+                              int reallocate_flag)
+{
+    VAStatus va_status = VA_STATUS_SUCCESS;
+    uint32_t fourcc = VA_FOURCC_NV12;
+    int update = 0;
+
+    if (!obj_surface) {
+        va_status = VA_STATUS_ERROR_INVALID_PARAMETER;
+
+        goto EXIT;
+    }
+
+    if (bit_depth_minus8 > 0) {
+        if (obj_surface->fourcc != VA_FOURCC_P010) {
+            update = 1;
+            fourcc = VA_FOURCC_P010;
+        }
+    } else if (obj_surface->fourcc != VA_FOURCC_NV12) {
+        update = 1;
+        fourcc = VA_FOURCC_NV12;
+    }
+
+    if (!obj_surface->bo || update) {
+        if (reallocate_flag) {
+            struct i965_driver_data * const i965 = i965_driver_data(ctx);
+
+            i965_destroy_surface_storage(obj_surface);
+
+            va_status = i965_check_alloc_surface_bo(ctx,
+                                                    obj_surface,
+                                                    i965->codec_info->has_tiled_surface,
+                                                    fourcc,
+                                                    SUBSAMPLE_YUV420);
+        } else
+            va_status = VA_STATUS_ERROR_INVALID_PARAMETER;
+    }
+
+EXIT:
+    return va_status;
+}
+
+static void
+hevc_get_max_mbps(uint32_t level_idc,
+                  uint32_t *max_bps,
+                  uint64_t *max_bps_per_pic)
+{
+    switch (level_idc) {
+    case 30:
+        *max_bps = 552960;
+        *max_bps_per_pic = 36864;
+        break;
+    case 60:
+        *max_bps = 3686400;
+        *max_bps_per_pic = 122880;
+        break;
+    case 63:
+        *max_bps = 7372800;
+        *max_bps_per_pic = 245760;
+        break;
+    case 90:
+        *max_bps = 16588800;
+        *max_bps_per_pic = 552760;
+        break;
+    case 93:
+        *max_bps = 33177600;
+        *max_bps_per_pic = 983040;
+        break;
+    case 120:
+        *max_bps = 66846720;
+        *max_bps_per_pic = 2228224;
+        break;
+    case 123:
+        *max_bps = 133693440;
+        *max_bps_per_pic = 2228224;
+        break;
+    case 150:
+        *max_bps = 267386880;
+        *max_bps_per_pic = 8912896;
+        break;
+    case 153:
+        *max_bps = 534773760;
+        *max_bps_per_pic = 8912896;
+        break;
+    case 156:
+        *max_bps = 1069547520;
+        *max_bps_per_pic = 8912896;
+        break;
+    case 180:
+        *max_bps = 1069547520;
+        *max_bps_per_pic = 35651584;
+        break;
+    case 183:
+        *max_bps = 2139095040;
+        *max_bps_per_pic = 35651584;
+        break;
+    case 186:
+        *max_bps = 4278190080;
+        *max_bps_per_pic = 35651584;
+        break;
+    default:
+        *max_bps = 16588800;
+        *max_bps_per_pic = 552760;
+        break;
+    }
+}
```

Comment:
Can this be moved to common code?  The bitrate level limits are already present in gen9_hevc_enc_utils.c.

Reply:
Definitely, it will be better to combine them together. but GEN10 needs max bit per picture which is not required by GEN9. I will check it more to decide how to deal it better.